axlearn/cloud/gcp/jobs/dataflow.py

```diff
@@ -243,13 +243,13 @@ def _execute(self):
             cmd = cfg.command
         else:
             cmd = (
-                "docker run --rm --entrypoint /bin/bash "
+                "docker run --rm --mount type=bind,src=$HOME/.config/gcloud,dst=/root/.config/gcloud --entrypoint /bin/bash "
                 f"{self._bundler.id(cfg.name)} -c '{cfg.command}'"
             )
         cmd = f"{cfg.setup_command} && {cmd}"
         cmd = f"bash -c {shlex.quote(cmd)}"
         logging.info("Executing in subprocess: %s", cmd)
-        with subprocess.Popen(cmd, shell=True, text=True) as proc:
+        with subprocess.Popen(cmd, shell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE) as proc:
```

Comment:
Does this mean the user will not be able to see the outputs in realtime?

Reply:
Before the change, I observed that stdout and stderr are always `None`. This is the documentation that suggested the change: https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate

`Note that if you want to send data to the process’s stdin, you need to create the Popen object with stdin=PIPE. Similarly, to get anything other than None in the result tuple, you need to give stdout=PIPE and/or stderr=PIPE too.`

However, for both before and after the change, users can't seem to see the outputs in realtime.

Reply:
Okay I tested this again and since after the change user can't see real time output, I reverted this change